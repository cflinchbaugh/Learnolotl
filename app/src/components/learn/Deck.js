import React, { Component } from 'react';
import CardWrapper from 'cards/CardWrapper';
import styled from 'styled-components';
import { connect } from 'react-redux';

const StyleWrapper = styled.div`
    margin: auto;
`

export class Deck extends Component {
    render() {
        let builtDeck = this._buildDeck(),
            deck = builtDeck.length ? builtDeck : "No Cards Loaded!";

        return (
            <StyleWrapper>
                <div className="deck-wrapper">
                    {deck}
                </div>
            </StyleWrapper>

        )
    }

    _buildDeck() {
        return this.props.deckData.cardData.map(function(cardData, idx) {
            return <CardWrapper 
                mode={this.props.deckData.mode}
                data={cardData}
                display={this.props.display}
                key={idx}/>
        }, this);
    }
}

const mapStateToProps = state => ({
    display: state.cards.flashCardDisplay
});

export default connect(mapStateToProps, { })(Deck);