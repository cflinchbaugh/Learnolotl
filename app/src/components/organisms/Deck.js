import React, { Component } from 'react';
import Card from '../molecules/Card';

class Deck extends Component {
    render() {
        let deck = this._buildDeck();

        if (deck.length) {
            return (
                <div className="deck-wrapper">
                    {deck}
                </div>
            );
        }
        return (
            <div className="deck-wrapper">
                Empty  
            </div>
        )
    }

    _buildDeck() {
        return this.props.deckData.cardData.map(function(cardData, idx) {
            return <Card 
                mode='hiragana'
                data={cardData}
                display={this.props.deckData.display}
                handleClickDisplay={this.props.deckData.handleClickDisplay}
                key={idx}/>
        }, this);
    }
}

export default Deck