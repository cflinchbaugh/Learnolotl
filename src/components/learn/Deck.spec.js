import React from 'react';
import {shallow} from 'enzyme';
import {Deck} from './Deck';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Deck', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const deckData = {
            deckData: {
                cardData: []
            }
        }

        const tree = renderer.render(
            <Deck {...deckData}/>
        );

        expect(tree).toMatchSnapshot();
    });
});