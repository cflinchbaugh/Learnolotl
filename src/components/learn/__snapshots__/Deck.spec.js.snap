// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck renders and matches snapshot 1`] = `
<styled.div
  className="col-xs-12"
>
  <div
    className="deck-wrapper"
  >
    No Cards Loaded!
  </div>
</styled.div>
`;
